File: project/hyeongmin-onboarding/src/main/java/kr/co/fastcampus/onboarding/hyeongminonboarding/domain/entity/SurveyResponse.java
package kr.co.fastcampus.onboarding.hyeongminonboarding.domain.entity;


import jakarta.persistence.*;
import kr.co.fastcampus.onboarding.hyeongminonboarding.global.entity.base.BaseTimeEntity;
import lombok.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;


/***
 * @author khm0813
 * - 설문 조사 - 응답 맵핑 테이블
 */
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "survey_response")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SurveyResponse extends BaseTimeEntity {
    @Id
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    private Survey survey;

    private int surveyVersion;

    private LocalDateTime submittedAt;
}

File: project/sungmin-onboarding/survey-service/src/main/kotlin/com/okdori/surveyservice/domain/SurveyResponse.kt
package com.okdori.surveyservice.domain

import com.okdori.surveyservice.domain.base.BaseTimeEntity
import io.hypersistence.utils.hibernate.id.Tsid
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

/**
 * packageName    : com.okdori.surveyservice.domain
 * fileName       : SurveyResponse
 * author         : okdori
 * date           : 2025. 6. 11.
 * description    :
 */

@Table(name = "t_survey_response")
@Entity
class SurveyResponse(
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    var survey: Survey
): BaseTimeEntity() {
    @Id @Tsid
    @Column(name = "response_id", columnDefinition = "CHAR(13)")
    var id: String? = null

    @Column(columnDefinition = "VARCHAR(100)")
    var responseUser: String? = null
}
